.github/workflows/bicep-build.yml

```diff
@@ -0,0 +1,92 @@
+## deploy azDeploySecureSub.bicep
+
+name: Azure REDCAP Deployment
+
+on:
+    workflow_dispatch:
+
+    push:
+        branches:
+            - main
+
+    # pull_request:
+    #     branches:
+    #         - main
+
+permissions:
+    id-token: write
+    contents: read
+
+env:
+#   bicepFile: ./azDeploySecureSub.bicep
+    azCliVersion: 2.5.72
+            
+jobs:
+    # Validate the Bicep templates
+    validateTemplates:
+        runs-on: ubuntu-latest
+        
+        steps:
+            - uses: action/checkout@v4
+              name: Checkout
+
+            - uses: azure/login@v1
+              name: Azure Login
+              with:
+                client-id: ${{ secrets.AZURE_CLIENT_ID }}
+                tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+                subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+            - uses: azure/cli@v1
+              name: Azure CLI
+              with:
+                azcliversion: ${{ env.azCliVersion }}
+                inlineScript: |
+                    az deployment sub validate --location westeurope --template-file ./azDeploySecureSub.bicep
+    
+    # validate the resources before creation
+    validateResources:
+        runs-on: ubuntu-latest
+        needs: [validateTemplates]
+        steps:
+            - uses: action/checkout@v4
+              name: Checkout
+
+            - uses: azure/login@v1
+              name: Azure Login
+              with:
+                client-id: ${{ secrets.AZURE_CLIENT_ID }}
+                tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+                subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+                client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+
+            - uses: azure/cli@v1
+              name: Azure CLI
+              with:
+                azcliversion: ${{ env.azCliVersion }}
+                inlineScript: |
+                    az deployment sub create --location westeurope --template-file ./azDeploySecureSub.bicep --what-if --verbose
+
+    # Deploy the resources
+    deployResources:
+        runs-on: ubuntu-latest
+        environment: 'DEMO-REDCAP'
+        needs: [validateTemplates, validateResources]
+        
+        steps:
+            - uses: action/checkout@v4
+              name: Checkout
+
+            - uses: azure/login@v1
+              name: Azure Login
+              with:
+                client-id: ${{ secrets.AZURE_CLIENT_ID }}
+                tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+                subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+            - uses: azure/cli@v1
+              name: Azure CLI
+              with:
+                azcliversion: ${{ env.azCliVersion }}
+                inlineScript: |
+                    az deployment sub create --location westeurope --template-file ./azDeploySecureSub.bicep --verbose
```

Comment:
Can't have a hardcoded location

Reply:
I think its fixed.

Reply:
OK, I'll approve this PR but note that this is still hardcoded.